testcases/LightPathDiagnostics.py

```diff
@@ -0,0 +1,253 @@
+            self.skipTest("FSP Platform OPAL LPD tests")
+        self.cv_FSP.fsp_get_console()
+        if not self.cv_FSP.mount_exists():
+            raise OpTestError("Please mount NFS and retry the test")
+
+        msg = "usysident is not supported on the PowerKVM Host platform"
+        res = self.cv_HOST.host_run_command("usysident")
+        self.assertNotIn(msg, res, "LightPathDiagnostics is not supported")
+
+    def get_location_codes(self):
+        console = self.cv_SYSTEM.sys_get_ipmi_console()
+        self.cv_SYSTEM.host_console_login()
+        self.cv_SYSTEM.host_console_unique_prompt()
+        console.run_command("uname -a")
+        res = console.run_command("usysident")
+        loc_codes = []
+        for loc in res:
+            loc_codes.append(loc.split("\t")[0])
+        return loc_codes
+
+    def get_sysattn_indicator_loc(self):
+        console = self.cv_SYSTEM.sys_get_ipmi_console()
+        self.cv_SYSTEM.host_console_login()
+        self.cv_SYSTEM.host_console_unique_prompt()
+        console.run_command("uname -a")
+        res = console.run_command("usysattn")
+        loc_codes = []
+        for loc in res:
+            loc_codes.append(loc.split("\t")[0])
+        return loc_codes[0]
+
+class UsysIdentifyTest(LightPathDiagnostics):
+
+    ##
+    # @brief This function tests usysident identification of LED's
+    #
+    def runTest(self):
+        self.lpd_init()
+        loc_codes = self.get_location_codes()
+        print "Executing indicator identify tests"
+        # Using -l option(Location code) Identifying a single device
+        for loc in loc_codes:
+            print "Turn on identification indicator %s from Host OS" % loc
+            self.cv_HOST.host_run_command("usysident -l %s -s identify" % (loc))
+            tries = 20
+            for i in range(1, tries+1):
+                response = self.cv_HOST.host_run_command("usysident -l %s" % (loc))
+                if "on" in response:
+                    break
+                time.sleep(1)
+            self.assertIn("on", response,
+                    "Turn ON of identification indicator %s is failed" % loc)
+            print "Current identification state of %s is ON" % loc
+
+        self.cv_HOST.host_run_command("usysident")
+
+        # Setting to normal state of device(Turn off Device)
+        for loc in loc_codes:
+            print "Turn off identification indicator %s from Host OS" % loc
+            self.cv_HOST.host_run_command("usysident -l %s -s normal" % (loc))
+            tries = 20
+            for i in range(1, tries+1):
+                response = self.cv_HOST.host_run_command("usysident -l %s" % (loc))
+                if "off" in response:
+                    break
+                time.sleep(1)
+            self.assertIn("off", response,
+                    "Turn OFF of identification indicator %s is failed" % loc)
+            print "Current identification state of %s is OFF" % loc
```

Comment:
usysident will use internally the sysfs paths only to do any LED operations. By this way we can test both sysfs and tool also.